### ISSUE TITLE
Migrate to version catalog

### ISSUE THREAD

**Issue by JakeWharton** [cashapp/molecule#103]

**Comment by Sripadmanabans** [comment]
I would like to assist with this. This would be my first contribution, please let me know if I have to follow some process to provide assistance. Thank you.

**Comment by JakeWharton** [comment]
Nope. Go for it! Feel free to check out other repos like Turbine for what we generally expect version catalog usage to look like.

**Comment by Sripadmanabans** [comment]
Looks like I don't have permission to push. Is there some way to request access?

I did look at [Turbine](https://github.com/cashapp/turbine) it did not seem to use version catalog. I did find [SQLDelight](https://github.com/cashapp/sqldelight) and tried to model it after it. Plugins also seem to have been moved to version catalog in that repository. Moving plugins to version catalog was proving trickier due to the usage of plugins in `subprojects`. Usage of `subprojects` seems to be discouraged according to this [issue](https://github.com/gradle/gradle/issues/18759), the recommendation is to use convention plugins. This would require more changes and I wanted to get your inputs before trying to make them.
